crates/core/src/logstore/storage/mod.rs

```diff
@@ -143,121 +100,22 @@ impl ObjectStoreRegistry for DefaultObjectStoreRegistry {
-
-/// TODO
-pub fn factories() -> FactoryRegistry {
-    static REGISTRY: OnceLock<FactoryRegistry> = OnceLock::new();
-    REGISTRY
-        .get_or_init(|| {
-            let registry = FactoryRegistry::default();
-            registry.insert(
-                Url::parse("memory://").unwrap(),
-                Arc::new(DefaultObjectStoreFactory::default()),
-            );
-            registry.insert(
-                Url::parse("file://").unwrap(),
-                Arc::new(DefaultObjectStoreFactory::default()),
-            );
-            registry
-        })
-        .clone()
+#[derive(Debug, Clone, Default)]
+pub struct LimitConfig {
+    pub max_concurrency: Option<usize>,
 }
 
-/// Simpler access pattern for the [FactoryRegistry] to get a single store
-pub fn store_for(url: &Url, storage_options: &StorageOptions) -> DeltaResult<ObjectStoreRef> {
-    let scheme = Url::parse(&format!("{}://", url.scheme())).unwrap();
-    if let Some(factory) = factories().get(&scheme) {
-        let (store, _prefix) = factory.parse_url_opts(url, storage_options)?;
-        Ok(store)
-    } else {
-        Err(DeltaTableError::InvalidTableLocation(url.clone().into()))
-    }
-}
-
-/// Options used for configuring backend storage
-#[derive(Clone, Debug, Serialize, Deserialize, Default)]
-pub struct StorageOptions(pub HashMap<String, String>);
-
-impl From<HashMap<String, String>> for StorageOptions {
-    fn from(value: HashMap<String, String>) -> Self {
-        Self(value)
-    }
-}
-
-/// Simple function to wrap the given [ObjectStore] in a [PrefixStore] if necessary
-///
-/// This simplifies the use of the storage since it ensures that list/get/etc operations
-/// start from the prefix in the object storage rather than from the root configured URI of the
-/// [ObjectStore]
-pub fn url_prefix_handler<T: ObjectStore>(store: T, prefix: Path) -> ObjectStoreRef {
-    if prefix != Path::from("/") {
-        Arc::new(PrefixStore::new(store, prefix))
-    } else {
-        Arc::new(store)
-    }
-}
-
-/// Simple function to wrap the given [ObjectStore] in a [LimitStore] if configured
-///
-/// Limits the number of concurrent connections the underlying object store
-/// Reference [LimitStore](https://docs.rs/object_store/latest/object_store/limit/struct.LimitStore.html) for more information
-pub fn limit_store_handler<T: ObjectStore>(store: T, options: &StorageOptions) -> ObjectStoreRef {
-    let concurrency_limit = options
-        .0
-        .get(storage_constants::OBJECT_STORE_CONCURRENCY_LIMIT)
-        .and_then(|v| v.parse().ok());
-
-    if let Some(limit) = concurrency_limit {
-        Arc::new(LimitStore::new(store, limit))
-    } else {
-        Arc::new(store)
-    }
-}
-
-/// Storage option keys to use when creating [ObjectStore].
-///
-/// The same key should be used whether passing a key in the hashmap or setting it as an environment variable.
-/// Must be implemented for a given storage provider
-pub mod storage_constants {
```

Comment:
i could not find any way we actually inspect the environment to pichk up this variable. that said, wit the updated configuration parsing we should now have a clear way how we can enable much more config via then environment in a manageable way.